sdk/test/opentelemetry/sdk/configurator_test.rb

```diff
@@ -167,7 +167,7 @@
       it 'defaults to NoopSpanProcessor if no valid exporter is available' do
         configurator.configure
 
-        _(OpenTelemetry.tracer_provider.active_span_processor).must_be_instance_of(
+        _(OpenTelemetry.tracer_provider.instance_variable_get(:@active_span_processor)).must_be_instance_of(
```

Comment:
This is the thing that I was not looking forward to with this change. Debugging / testing gets harder here, but I suppose it's still a good idea.

Reply:
Agreed. In general, though, nothing outside of the SDK should know or care about the specific TracerProvider that's installed. I don't have the energy to do it now, but I think we could clean up these tests to stub the appropriate class's `on_span_start` and start a span instead.

Reply:
>  I don't have the energy to do it now

No need, either. That sounds like good rainy-day refactoring for someone who feels up to it, definitely not necessary at all! 😄